src/mscorlib/shared/System/ReadOnlySpan.cs

```diff
@@ -212,11 +218,13 @@ public void CopyTo(Span<T> destination)
         /// <param name="destination">The span to copy items into.</param>
         public bool TryCopyTo(Span<T> destination)
         {
-            if ((uint)_length > (uint)destination.Length)
-                return false;
-
-            Span.CopyTo<T>(ref destination.DangerousGetPinnableReference(), ref _pointer.Value, _length);
-            return true;
+            bool retVal = false;
+            if ((uint)_length <= (uint)destination.Length)
+            {
+                Span.CopyTo<T>(ref destination.DangerousGetPinnableReference(), ref _pointer.Value, (nuint)_length);
+                retVal = true;
+            }
+            return retVal;
```

Comment:
This results in better code gen?  If so, why can't the JIT just do that rather than requiring it be written one way or the other?
cc: @AndyAyersMS